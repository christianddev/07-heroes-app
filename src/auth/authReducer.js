import {types} from '../types/types';

// const statec = {
//     name: 'David',
//     logged: true
// };

const authReducer = (user = {}, action)=>  {

    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }
        case types.logout:
            return {
                logged: false
            }
        default:
            return user;
    }
}
export default authReducer;