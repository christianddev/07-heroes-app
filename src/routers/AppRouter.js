import React, { useContext } from "react";
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { AuthContext } from "../auth/AuthContext";
import PrivateRoute from "./PrivateRoute";
import PublicRoute from "./publicRoute";

import LoginScreen from "../components/login/LoginScreen";
import DashboardRoutes from "./DashboardRoutes";

const AppRouter = () => {
  const {user:{logged}} = useContext(AuthContext);
    return ( 
        <Router>
          <div>
            <Switch>
              <PublicRoute 
                exact
                path="/login" 
                component={LoginScreen}
                isAuthenticated={logged}
              />
              <PrivateRoute 
                path="/" 
                component={DashboardRoutes} 
                isAuthenticated={logged}
              />
            </Switch>
        </div>
      </Router>
    )
}

export default AppRouter;
