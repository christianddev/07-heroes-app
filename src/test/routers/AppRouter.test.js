import React from 'react';
import { mount } from "enzyme"
import { AuthContext } from "../../auth/AuthContext"
import AppRouter from "../../routers/AppRouter"
import { userMockLogged } from '../fixtures/userMock';

describe('AppRouter.js', () => {

  const contextvalue = {
    dispath: jest.fn(),
    user: {
      logged: false,
      name: ''
    }
  }

  test('should show Login if user has not authenticated', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextvalue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  })

  test('should show Marvel Component if user has authenticated', () => {
    const contextvalue = {
      dispath: jest.fn(),
      user: userMockLogged
    }
    const wrapper = mount(
      <AuthContext.Provider value={contextvalue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    expect(wrapper.find('.navbar').exists()).toBeTruthy();
  })
})
