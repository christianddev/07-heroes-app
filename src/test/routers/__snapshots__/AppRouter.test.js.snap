// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter.js should show Login if user has not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
